AutoGrade/grader.py

```diff
@@ -88,6 +88,66 @@ def run_student_tests(target_folder, total_points, timeout):
     out_file =  os.path.join(target_folder, "test-results" + ".log")
     touch(out_file)
 
+
+    """
+    # This code can be used to execute and compile code throught compile, In reference to PR "Languagespp - Temporary Record"
+
+        # Compiling
+        logger.debug("Checking code for compilation error")
+
+        # We need to bring assignment variable here.
+        cmd = assignment.language.compile_cmd
+        cmd = process_cmd(cmd, mainfile, timeout, basename)
+        cmd = cmd.split(" ")
+
+        process = Popen(cmd, cwd=target_folder, stdout=f, stderr=f, shell=True)    
+                
+        (output, err) = process.communicate()
+        exit_code = process.wait()
+
+        if exit_code == 0:
+            # Executing
+            logger.debug("Executing code against test cases")
+
+            cmd = assignment.language.execution_cmd
+            cmd = process_cmd(cmd, mainfile, timeout, basename)
+            cmd = cmd.split(" ")
+
+            # TODO: Check against pytest, STDIN, File
+            # Problem: We will need to add two files, in and out...
+            process = Popen(cmd, cwd=target_folder, stdout=f, stderr=f, shell=True)    
+                        
+            (output, err) = process.communicate()
+            p.start()
+            exit_code = process.wait()
+        else:
+            logger.debug("Code failed compilation error")          
+            
+    with open(out_file) as f:
+        out = f.read()
+
+    # If it is pytest then read score from log file
+    if "pytest" in assignment.language.name.lower():
+        res_line = out.splitlines()[-1]
+        try:
+            score = get_score_from_result_line(res_line, total_points)
+        except EnvironmentError:
+            logger.error("===== EnvironmentError found. ======== ")
+            logger.error(out)
+            out = "Your assignment had critical errors. Please review. If you think this is a system issue, please send details to your admin.\n" + out
+            # write back to the out file
+            with open(out_file, 'w') as f:
+                f.write(out)
+            score = (0, 0) # error means you get a 0
+
+        logger.debug("Read test line [" + res_line.strip("=") + "]")
+        logger.debug("Calculated score: " + str(score))
+    else:
+        pass
```

Comment:
We have problem here... how to supply input to the code, through FILE, STDIN or ARGS? not decided yet (This is crucial).